tests/cpptests/test_cpp_thpool.cpp

```diff
@@ -147,3 +147,60 @@ TEST_F(PriorityThpoolTestWithoutPrivilegedThreads, CombinationTest) {
     ASSERT_LT(arr[2], arr[4]);
     ASSERT_LT(arr[4], arr[3]);
 }
+
+/* ========================== NUM_THREADS = 2, NUM_PRIVILEGED = 0 ========================== */
+THPOOL_TEST_SUITE(PriorityThpoolTestFunctionality, 1, 0)
+
+void sleep_1_and_set(size_t *time_ms) {
+    std::this_thread::sleep_for(std::chrono::milliseconds(*time_ms));
+}
+
+TEST_P(PriorityThpoolTestFunctionality, TestTerminateWhenEmpty) {
+    // Pause the thread pool before adding tasks, to validate that jobs won't get executed before
+    // all other jobs are inserted into the queue.
+    redisearch_thpool_terminate_pause_threads(this->pool);
```

Comment:
Use this test to ensure that the thread pool is in the expected state after pushing the admin jobs.